### PR TITLE
Add new indexer Wolfmax4K

### LINKED ISSUE #14511: [req]: Wolfmax4k
### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Is there already a request for your tracker?

- [X] I have checked older issues, open and closed

### Type

Public

### Tracker details

```markdown
- Name: Wolfmax4k
- URL: https://wolfmax4k.com/
- Description: Wolfmax4k is a SPANISH Public tracker for MOVIES / TV
- Minimum Seeding Time and Minimum Ratio (if Private): 
- Additional Comments: 
```

### PR DESCRIPTION
**Pull request by highfredo**
#### Description
Add new indexer Wolfmax4K

#### Issues Fixed or Closed by this PR

* Fixes #14511

### CODE REVIEW

File: src/Jackett.Common/Indexers/Wolfmax4K.cs [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+namespace Jackett.Common.Indexers
+{
+    [ExcludeFromCodeCoverage]
+    public class Wolfmax4K : IndexerBase
+    {
+        public override string Id => "wolfmax4k";
+        public override string Name => "Wolfmax 4k";
+        public override string Description => "Wolfmax 4k is a SPANISH public tracker for MOVIES / TV";
+        public override string SiteLink { get; protected set; } = "https://wolfmax4k.com/";
+        public override string[] AlternativeSiteLinks => new[]
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/Wolfmax4K.cs:30]: if there is only one domain then you don not need to include an alternative site links block ;-)

File: src/Jackett.Common/Indexers/Wolfmax4K.cs [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+            // https://wolfmax4k.com/descargar/la-sala-de-torturas-chinas/
+            // https://wolfmax4k.com/pelicula/el-hombre-de-chinatown/
+            // https://wolfmax4k.com/descargar/documentales/misterios-de-china/temporada-1/capitulo-03/
+            // https://wolfmax4k.com/descargar/serie/the-legend-of-vox-machina/temporada-2/capitulo-11/
+            // https://wolfmax4k.com/descargar/peliculas-x264-mkv/los-tesoros-del-mar-de-china-1987-/bluray-microhd/
+            // https://wolfmax4k.com/descargar/serie/the-legend-of-vox-machina/temporada-1/capitulo-04-al-06/
+            // https://wolfmax4k.com/descargar/serie-en-hd/top-boy/temporada-3/capitulo-02/
+            // https://wolfmax4k.com/descargar/programas-tv/la-isla-de-las-tentaciones/temporada-7/capitulo-10/
+            // https://wolfmax4k.com/descargar/serie-1080p/historial-delictivo/temporada-1/capitulo-02/
+
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/Wolfmax4K.cs:244]: some torznab apps do not like getting results with 0 byte sizes. If the site does not provide a size then we usually guess some default size values 
https://github.com/Jackett/Jackett/blob/7c49cea4b64cc95166250d441d94f1692eb00dcf/src/Jackett.Common/Indexers/DonTorrent.cs#L505-L516
https://github.com/Jackett/Jackett/blob/7c49cea4b64cc95166250d441d94f1692eb00dcf/src/Jackett.Common/Indexers/DonTorrent.cs#L527-L541

**Review by garfield69 — CHANGES_REQUESTED** [review]
Hey, thanks for you contribution!
While I am not a C# dev, I have made some suggestions based on the usual Jackett indexer ethos. I'll get a proper C# to review the code later.

**Comment by garfield69** [comment]
so the site does not support searching for season and episodes `saving hope s02e01` or even `saving hope temporada 2  capitulo 10` ?
this may cause issues with Sonarr. We may need to strip the season and episode from the query and filter results.

**Comment by highfredo** [comment]
> so the site does not support searching for season and episodes `saving hope s02e01` or even `saving hope temporada 2 capitulo 10` ? this may cause issues with Sonarr. We may need to strip the season and episode from the query and filter results.

Thanks for your feedback!
As soon as I can I will make the changes

Unfortunately the site does not support searches by episodes, and the filter by categories does not work correctly. 
But the indexer try to fix it [here](https://github.com/highfredo/Jackett/blob/c53332c1dc2208884e104ebf7edb1597ee267931/src/Jackett.Common/Indexers/Wolfmax4K.cs#L260-L281)

So, Sonarr search works :)
image

**Comment by highfredo** [comment]
> Search for S01E02

S01E02 works, but S01E03 fails 🤦

File: src/Jackett.Common/Indexers/Wolfmax4K.cs [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+        {
+            public static string Pelicula => "pelicula";
+            public static string Pelicula720 => "pelicula720";
+            public static string Pelicula1080 => "pelicula1080";
+            public static string Pelicula4K => "pelicula4k";
+            public static string Serie => "serie";
+            public static string Serie720 => "serie720";
+            public static string Serie1080 => "serie1080";
+            public static string Serie4K => "serie4k";
+        }
```
[the remark this answers is not in the document]
Comment (highfredo) [reply src/Jackett.Common/Indexers/Wolfmax4K.cs:49]: Are you referring to a new class within Wolfmax4K.cs or in its own Wolfmax4KCatType.cs file?

If it is in your own file, where should I place it? I don't see anything similar. In fact, the indexers that I have looked at (dontorrent and mejortorrent) have it like me.

**Review by garfield69 — APPROVED** [review]
Tested, LGTM

**Pull request merged**